strawberryfields/backends/tfbackend/states.py

```diff
@@ -170,41 +121,30 @@ def all_fock_probs(self, **kwargs):
             probs[0,2,3]
 
         Args:
-            **kwargs: Optional keyword arguments.
-
-                * If this contains the key ``eval``, then the corresponding argument will be used to determine the return behaviour of this function.
-                  When ``eval=True``, the return value is numerical; when ``eval=False``, it is symbolic.
-                * If eval is not present in kwargs, then state falls back to the an internal evaluation behaviour,
-                  which is specified at initialization.
-                * A Tensorflow Session or feed_dict may also be passed via the keys ``session`` or ``feed_dict``, respectively.
-                  If a Session is supplied, then ``eval`` is overriden and the numerical evaluation takes place in the provided Session.
-                  If session and/or feed_dict are not given, then a temporary session and/or empty feed_dict will be used.
+
         Returns:
             array/Tensor: the numerical values, or an unevaluated Tensor object, for the Fock-basis probabilities.
         """
 
-        with self.graph.as_default():
-
-            state = self.data
-            if not self.batched:
-                state = tf.expand_dims(state, 0) # add fake batch dimension
+        state = self.data
+        if not self.batched:
+            state = tf.expand_dims(state, 0) # add fake batch dimension
 
-            if self.is_pure:
-                probs = tf.abs(state) ** 2
-            else:
-                # convert to index scheme compatible with tf.diag_part
-                rng = np.arange(1, 2 * self.num_modes + 1, 2)
-                perm = np.concatenate([[0], rng, rng + 1])
-                perm_state = tf.transpose(state, perm)
-                probs = tf.map_fn(tf.diag_part, perm_state) # diag_part doesn't work with batches, need to use map_fn
+        if self.is_pure:
+            probs = tf.abs(state) ** 2
+        else:
+            # convert to index scheme compatible with tf.linalg.diag_part
+            rng = np.arange(1, 2 * self.num_modes + 1, 2)
+            perm = np.concatenate([[0], rng, rng + 1])
+            perm_state = tf.transpose(state, perm)
+            probs = tf.map_fn(tf.linalg.tensor_diag_part, perm_state) # diag_part doesn't work with batches, need to use map_fn
 
-            if not self.batched:
-                probs = tf.squeeze(probs, 0)  # drop fake batch dimension
+        if not self.batched:
+            probs = tf.squeeze(probs, 0)  # drop fake batch dimension
 
-            probs = tf.identity(probs, name="all_fock_probs")
-            probs = self._run(probs, **kwargs)
+        probs = tf.identity(probs, name="all_fock_probs")
 
-            return probs
+        return probs.numpy()
```

Comment:
Unlike the measurements, where casting back to numpy is probably fine, here I feel like this might break differentiability?

Reply:
If I remember correctly also in this case the tests would not pass without the call to .numpy().
I can try again and perhaps modify the tests (i.e. call .numpy() there for example)?